src/app/client/src/app/modules/dial-code-search/services/dial-code/dial-code.service.ts

```diff
@@ -1,36 +1,66 @@
-import { ConfigService } from '@sunbird/shared';
+import { ConfigService, ServerResponse} from '@sunbird/shared';
 import { SearchService, PlayerService } from '@sunbird/core';
+import { UserService } from '../../../../modules/core/services/user/user.service';
 import { Injectable } from '@angular/core';
 import * as _ from 'lodash-es';
 import { map, catchError, retry } from 'rxjs/operators';
 import { of, Observable, iif, forkJoin, empty } from 'rxjs';
 import * as TreeModel from 'tree-model';
+import { PublicDataService } from '../../../../modules/core/services/public-data/public-data.service';
 const treeModel = new TreeModel();
 
 @Injectable({
   providedIn: 'root'
 })
 export class DialCodeService {
 
+  /**
+   * Reference of config service
+   */
+  public config: ConfigService;
+
+  /**
+   * Reference of user service.
+   */
+  public user: UserService;
+
+  /**
+   * Reference of public data service
+   */
+  public publicDataService: PublicDataService;
 
   private dialSearchResults;
-  constructor(private searchService: SearchService, private configService: ConfigService, private playerService: PlayerService) { }
+  constructor(private searchService: SearchService, private configService: ConfigService, private playerService: PlayerService,
+    config: ConfigService, user: UserService, publicDataService: PublicDataService) {
```

Comment:
do this in constructor param, private config: ConfigService, private user: UserService, private publicDataService: PublicDataService and remove below lines 

  /**
   * Reference of config service
   */
  public config: ConfigService;

  /**
   * Reference of user service.
   */
  public user: UserService;

  /**
   * Reference of public data service
   */
  public publicDataService: PublicDataService;

      this.config = config;
      this.user = user;
      this.publicDataService = publicDataService;